src/game/systems/command.rs

```diff
@@ -468,13 +473,15 @@ pub fn command_system(
                             target_position,
                             mut target_dropped_item,
                             target_owner,
+                            target_expire_time,
                         )) = is_valid_pickup_target(
                             position,
                             target_entity,
                             &mut pickup_dropped_item_target_query,
                         ) {
-                            let result = if !target_owner
-                                .map_or(true, |owner| owner.entity == entity)
+                            let result =
+                            if !target_owner.map_or(true, |owner| owner.entity == entity) &&
+                                !target_expire_time.map_or(false, |expire_time| server_time.now >= expire_time.when)
```

Comment:
Instead of checking the expire time here, I think it would be more "ECS like" to have an owner expiry system which removes the Owner component from the dropped entity.

Similiar to how the expire_time_system works in src/game/systems/expire_time.rs

Maybe rename expire_time_system to expire_entity_system, which operates on the EntityExpireTime component, then add a new system called expire_owner_system which runs an Entity with Owner and OwnerExpireTime components and when the expire time is hit it just removes the Owner component from the entity.